jdaviz/configs/cubeviz/helper.py

```diff
@@ -85,7 +85,10 @@ def select_wavelength(self, wavelength):
             wavelength = float(wavelength.wavelength)
         if not isinstance(wavelength, (int, float)):
             raise TypeError("wavelength must be a float or int")
-        x_all = self.app.get_viewer('spectrum-viewer').data()[0].spectral_axis.value
+        # Retrieve the x slices from the spectrum viewer's marks
+        x_all = [m for m in self.app.get_viewer('spectrum-viewer').figure.marks
```

Comment:
I am not familiar with marks. Is it always in the unit we think it is in, especially with all the unit conversion going on via plugin, etc?

Reply:
the marks are in the plotted units, the click/drag even is guaranteed to be in plotted units, and the `select_wavelength` method says that it assumes the user input for wavelength is provided in plotted units, so I think we're safe (for now).

Reply:
I think the marks will always be in the same units of the spectrum viewer, since they are the literal marks that are plotted on screen. When using the slice tool, it will naturally request the wavelength in units of the viewer it's acting on, so I think that should be consistent? @kecnry is that a fair statement?

Reply:
Yes, I think that's fair.  The only possible problem I can see would be when the user calls the API manually, but the API docs state that no unit conversion is done, so I think that can be kicked down the road for the unit conversion refactor (if at all)